cpp/cmake_modules/FindArrow.cmake

```diff
@@ -95,6 +101,13 @@ if (MSVC)
     PATH_SUFFIXES "bin" )
   get_filename_component(ARROW_SHARED_LIBS ${ARROW_SHARED_LIBRARIES} PATH )
   get_filename_component(ARROW_PYTHON_SHARED_LIBS ${ARROW_PYTHON_SHARED_LIBRARIES} PATH )
+
+  if (PYARROW_BUILD_ARROW_GPU)
```

Comment:
I'm not sure PYARROW variables should be depended on in the main CMake files.

Reply:
Agreed. Probably want to add a `FindArrowCuda.cmake` file

Reply:
I agree (I now noticed that `python/cmake_modules` is actually a link to `cpp/cmake_modules`).
I'll fix it.

Reply:
Fixed.